mantidproject/mantid
# Show correct attribute values in Indirect Data Analysis FunctionBrowser

**Description of work.**
This PR adds a missing functions to manipulate attributes in `IFunctionModel` and `IFunctionView`. Both classes were missing `getAttribute` and `setAttribute` methods. Due to changes made in a previous PR, these can now be dealt with in the exact same manner as `setParameter(..)`.

The primary motivation for this work was to show the correct `Q` value for `Q` dependent functions in the `ConvFit` tab of the indirect data analysis GUI, as such as the testing instructions will relate to this issue:

**Test instructions**

1. Open the Indirect data analysis GUI
2. Go to Conv fit and load data from #26631
3. Add the fit type as the TeixeiraWater function
4. Change to full function view, you should note that the correct `Q` value is shown
5. Switch between spectra and the Q value will change
6. Change the `Q` value and switch on/off the full function view, you should note that the `Q` value that you set remained – signifying that the model has been appropriately updated.

As an aside, `Q` values set in this browser are not used in the Convolution fit performed in this Tab, as the 'Q' values are fetched from the workspace on-the-fly when a fit is requested (which explains why the correct values were used before even when the browser showed the incorrect Q values). Fixing this issue (if desired) would require another PR.

Fixes #28529 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
